libsolidity/analysis/DeclarationTypeChecker.cpp

```diff
@@ -146,7 +146,9 @@ void DeclarationTypeChecker::endVisit(UserDefinedTypeName const& _typeName)
 		_typeName.annotation().type = TypeProvider::enumType(*enumDef);
 	else if (ContractDefinition const* contract = dynamic_cast<ContractDefinition const*>(declaration))
 		_typeName.annotation().type = TypeProvider::contract(*contract);
-	else
+	// Can't assign the type for modifiers at this stage as it needs info from
```

Comment:
I'm actually not sure we should even get here. This is really the context of a type. Maybe we should split this into UserDefinedTypeName and `Path` (which is just a list of identifiers separated by a dot instead of something really signifying a type)

Reply:
Well, the UserDefinedTypeName is a parser construct which can't differentiate between those too..

Reply:
Actually I think the reason we got into this problem is that modifiers are much closer to the expression context than to the type context. So this should maybe even be a MemberAccess instead.